pkg/component/server/konnectivity.go

```diff
@@ -188,28 +188,28 @@ func (k *Konnectivity) runLease() {
 		}()
 		for {
 			select {
+			case <-k.done:
+				return
 			case <-ticker.C:
 				observedLeaseHolders := k.leaseLock.CountValidLeaseHolders(ctx)
 				if observedLeaseHolders != k.serverCount {
 					logrus.Debugf("change in %v lease holders detected. refreshing service.", serviceName)
 					k.serverCount = observedLeaseHolders
 
 					// restarting service
-					k.Stop()
-					k.Run()
+					if err := k.Stop(); err != nil {
+						logrus.Errorf("error stopping %v service: %v", serviceName, err)
+					}
+					if err := k.Run(); err != nil {
+						logrus.Errorf("error starting %v service: %v", serviceName, err)
+					}
```

Comment:
If `k.Stop()` here above fails (means that konnectivity is still running), should we really start another instance? Would it be better to have an `else` there, so we only do `k.Run()` if the `k.Stop()` was successful?

Reply:
oh, right. so instead of dealing with the level of `k.Stop()` and `k.Run()` we should rather stop and start only the supervisor (konnectivity-server process only).

Reply:
either that or we implement the `runLease()` thingy as a separate Component, that is conditionally added from `cmd/server.go` and passes in the konnectivity component.